src/hdf5/read_dpr_gpm.py

```diff
@@ -0,0 +1,162 @@
+    # get all the dimensions
+    dims = {}
+    data = {}
+    for k in h5ls.names:
+        if 'AlgorithmRuntimeInfo' in k:
+            continue
+
+        if 'DimensionNames' in df[k].attrs:
+            data[k] = df[k][:]
+            dims[k] = df[k].attrs['DimensionNames'].decode('UTF-8').split(',')
+    df.close()
+
+    unique_dims = {}
+    for k in data:
+        shape = data[k].shape
+        for idim, dname in enumerate(dims[k]):
+            # xarray does not allow method as dimension for whatever reason
+            dname = dname.replace('method', 'method1')
+            unique_dims[dname] = np.arange(shape[idim])
+
+    xr_data = xr.Dataset()
+    for dname in unique_dims:
+        xr_data[dname] = unique_dims[dname]
+
+    for k in data:
+        xr_dims = OrderedDict()
+        for dname in dims[k]:
+            dname = dname.replace('method', 'method1')
+            xr_dims[dname] = unique_dims[dname]
+        xr_data[k] = xr.DataArray(data[k], dims=xr_dims)
+
+    epoch_time = np.zeros(xr_data.nscan.size)
+    for iscan in xr_data.nscan.values:
+        time1 = datetime.datetime(xr_data['FS/ScanTime/Year'].values[iscan],
+                                  xr_data['FS/ScanTime/Month'].values[iscan],
+                                  xr_data['FS/ScanTime/DayOfMonth'].values[iscan],
+                                  xr_data['FS/ScanTime/Hour'].values[iscan],
+                                  xr_data['FS/ScanTime/Minute'].values[iscan],
+                                  xr_data['FS/ScanTime/Second'].values[iscan],
+                                  xr_data['FS/ScanTime/MilliSecond'].values[iscan])
+        epoch_time[iscan] = round((time1 - epoch).total_seconds())
+
+    # build output dictionary
+    dprdata = xr.Dataset()
+    dprdata['lat'] = xr_data['FS/Latitude']       # "nscan,nray=nfov";
+    dprdata['lon'] = xr_data['FS/Longitude']      # "nscan,nray=nfov";
+    dprdata['height'] = xr_data['FS/PRE/height']  # "nscan,nray=nfov,nbin=nelev";
+    dprdata['epoch_time'] = xr_data['FS/navigation/scLat'].copy()
+    dprdata['epoch_time'].values = np.squeeze(epoch_time)
+    dprdata['obs_measured'] = xr_data['FS/PRE/zFactorMeasured']  # "nscan,nray=nfov,nbin=nelev,nfreq=nchan"
+    dprdata['obs'] = xr_data['FS/SLV/zFactorFinal']  # "nscan,nray=nfov,nbin=nelev,nfreq=nchan"
+    dprdata['obs'].values[dprdata['obs'].values < -100] = np.nan
+    dprdata['zenith_angle'] = xr_data['FS/PRE/localZenithAngle']  # nscan,nray=nfov,nfreq=nchan
+    dprdata['azimuth_angle'] = dprdata['zenith_angle'].copy() * 0
+    dprdata['centerFreq'] = xr.DataArray(np.array([13.6, 35.5]), dims={"nfreq": 2})  # GHz
+    dprdata['centerWN'] = xr.DataArray(np.array([0.453, 1.183]), dims={"nfreq": 2})  # 1/cm
+
+    # rename dimensions
+    dprdata = dprdata.rename({'nbin': 'elevation', 'nfreq': 'channel', 'nray': 'fov', 'nscan': 'scanline'})
+    dprdata['fov1'] = dprdata.fov.copy()
+
+    return dprdata
+
+
+def read_dpr_gpm(dpr_fnames,
```

Comment:
this function is only used by this decoder and scans for all files

generally like to handle this outside the decoder, pass in wildcards or a list of files that one would like to process